query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -154,14 +155,27 @@ private String convertFilter2String(
           break;
         case CONTAINS_KEY:
           LiteralConstant[] kvp = convertExpressionToMapLiterals(filter.getRhs());
-          builder.append(convertExpressionToMapKeyColumn(filter.getLhs()));
+          builder.append(convertExpressionToMapKeyColumn(getLogicalColumnName(filter.getLhs())));
```

Comment:
I think we're doing the opposite of what we want here. CONTAINS_KEY and CONTAINS_KEYVALUE operators are defunct - attribute expressions replace them, and it doesn't make sense to use the two together. In both of these operators, the path is specified as an operator argument (RHS) - attribute expression moves path to the LHS in order to allow it to be used with any operator. These two special cased ones would go away, since it would never make sense to have something like `AttributeExpression("myCol", "my.path") CONTAINS_KEY_VALUE ("my.path", "my.value")` - we'd simply say `AttributeExpression("myCol", "my.path") EQUALS "my.value"`

Reply:
resolved